hk_install_centos7.sh

```diff
@@ -0,0 +1,289 @@
+yum versionlock golang-*
+
+# Uncomment these two lines if newer than 1.9.3 becomes supported but not 1.10.x
+#goVersionNumber=`yum --showduplicates list golang |grep -oP '1.9\.[0-9].*centos?' | sort |tail -n 1`
+#yum install -y golang-$goVersionNumber
+
+# Uncomment this line if 1.10.x ever becomes supported
+#yum install -y golang
+
+# Create a user for hooknode and setup go
+adduser hooknode
+
+su - hooknode -c "echo \"export GOROOT=/usr/lib/golang\" >> /home/hooknode/.bash_profile"
+su - hooknode -c "echo \"export GOPATH=/home/hooknode/go\" >> /home/hooknode/.bash_profile"
+su - hooknode -c "echo \"export PATH=/home/hooknode/go/bin:\$PATH\" >> /home/hooknode/.bash_profile"
+
+su - hooknode -c "mkdir -p /home/hooknode/go/src"
+su - hooknode -c "mkdir -p /home/hooknode/go/bin"
+
+# Install the go dependancy manager
+su - hooknode -c "go get -u github.com/golang/dep/cmd/dep"
+
+# Pull the hooknode repo
+su - hooknode -c "mkdir -p /home/hooknode/go/src/github.com/oysterprotocol"
+su - hooknode -c "cd /home/hooknode/go/src/github.com/oysterprotocol; git clone https://github.com/oysterprotocol/hooknode.git"
+su - hooknode -c "cp /home/hooknode/go/src/github.com/oysterprotocol/hooknode/.env.example /home/hooknode/go/src/github.com/oysterprotocol/hooknode/.env"
+su - hooknode -c "cd /home/hooknode/go/src/github.com/oysterprotocol/hooknode/; make install-deps"
+su - hooknode -c "cd /home/hooknode/go/src/github.com/oysterprotocol/hooknode/; go build -o ./bin/main.go"
+
+# Setup systemd service
+cat <<EOF | tee /lib/systemd/system/hooknode.service
+[Unit]
+Description=Oyster Hooknode in Golang
+After=network.target
+[Service]
+WorkingDirectory=/home/hooknode/go/src/github.com/oysterprotocol/hooknode
+User=hooknode
+PrivateDevices=yes
+ProtectSystem=full
+Type=simple
+ExecReload=/bin/kill -HUP $MAINPID
+KillMode=mixed
+KillSignal=SIGTERM
+TimeoutStopSec=60
+ExecStart=/home/hooknode/go/src/github.com/oysterprotocol/hooknode/bin/main.go
+SyslogIdentifier=HOOKN
+Restart=on-failure
+RestartSec=30
+[Install]
+WantedBy=multi-user.target
+Alias=hooknode.service
+EOF
+
+# Start service
+service hooknode start
+systemctl start hooknode.service
+systemctl enable hooknode.service
+
+# Start iota again
+service iota start
+
+
+# Open the firewall for incoming connections
+firewall-cmd --zone=public --add-port=14265/tcp --permanent
+firewall-cmd --zone=public --add-port=14265/udp --permanent
+firewall-cmd --zone=public --add-port=14600/udp --permanent
+firewall-cmd --zone=public --add-port=15600/tcp --permanent
+firewall-cmd --reload
+
+
+# Get IPs
+externalIps="$(dig +short myip.opendns.com @resolver1.opendns.com)"
+internalIps="$(hostname -I)"
+internalIps="$(echo -e "${internalIps}" | tr -d '[:space:]')"
+
+# Prepare and show confirmation message
+echo "Installation complete!"
+echo ""
+echo "You will now need to open the appropriate ports on your firewall/router:"
+echo "Port 14265 - TCP and UDP"
+echo "Port 14600 - UDP"
+echo "Port 15600 - TCP"
+echo ""
```

Comment:
It's not crucial, but you should probably add port 18600 (TCP) for Nelson. You can add a comment to open the ports on the router if they want to check on their node from outside. I think you don't need to open the ports in the router if you're just calling localhost from inside.

Reply:
Actually, close down 14265 - it's unsafe to leave it exposed and the Go server will interact with API using localhost.